common/persistence/data_store_interfaces_test.go

```diff
@@ -0,0 +1,177 @@
+		assertEmpty := func(b []byte, s string) {
+			assert.Nil(t, b, "should have nil bytes")
+			assert.Empty(t, s, "should have empty string")
+		}
+		assertEmpty(FromDataBlob(nil))
+		assertEmpty(FromDataBlob(&DataBlob{}))
+		assertEmpty(FromDataBlob(&DataBlob{Encoding: common.EncodingTypeThriftRW}))
+
+		dat, str := FromDataBlob(&DataBlob{
+			Data:     []byte("data"),
+			Encoding: common.EncodingTypeJSON,
+		})
+		assert.Equal(t, []byte("data"), dat, "data should be returned")
+		assert.Equal(t, string(common.EncodingTypeJSON), str, "encoding should be returned as a string")
+	})
+	t.Run("ToNilSafeDataBlob", func(t *testing.T) {
+		orig := &DataBlob{Encoding: common.EncodingTypeGob} // anything not empty
+		assert.Equal(t, orig, orig.ToNilSafeDataBlob())
+		assert.NotNil(t, (*DataBlob)(nil).ToNilSafeDataBlob(), "typed nils should convert to non-nils")
+	})
+	t.Run("GetEncodingString", func(t *testing.T) {
+		assert.Equal(t, "", (*DataBlob)(nil).GetEncodingString())
+		assert.Equal(t, "test", (&DataBlob{Encoding: "test"}).GetEncodingString())
+	})
+	t.Run("GetData", func(t *testing.T) {
+		// this method returns empty slices, not nils.
+		// I'm not sure if this needs to be maintained, but it must be checked before changing.
+		assert.Equal(t, []byte{}, (*DataBlob)(nil).GetData())
+		assert.Equal(t, []byte{}, (&DataBlob{}).GetData())
+		assert.Equal(t, []byte{}, (&DataBlob{Data: []byte{}}).GetData())
+		assert.Equal(t, []byte("test"), (&DataBlob{Data: []byte("test")}).GetData())
+	})
+	t.Run("GetEncoding", func(t *testing.T) {
+		same := func(encoding common.EncodingType) {
+			assert.Equal(t, encoding, (&DataBlob{Encoding: encoding}).GetEncoding())
+		}
+		unknown := func(encoding common.EncodingType) {
+			assert.Equal(t, common.EncodingTypeUnknown, (&DataBlob{Encoding: encoding}).GetEncoding())
+		}
+		// obvious
+		same(common.EncodingTypeGob)
+		same(common.EncodingTypeJSON)
+		same(common.EncodingTypeThriftRW)
+		same(common.EncodingTypeEmpty)
+
+		// highly suspicious
+		unknown(common.EncodingTypeProto)
+
+		// should be unknown
+		unknown(common.EncodingTypeUnknown)
+		unknown("any other value")
+	})
+	t.Run("To and FromInternal", func(t *testing.T) {
+		data := []byte("some data")
+		// safe
+		internal := &types.DataBlob{
+			EncodingType: types.EncodingTypeJSON.Ptr(),
+			Data:         data,
+		}
+		blob := &DataBlob{
+			Encoding: common.EncodingTypeJSON,
+			Data:     data,
+		}
+		assert.EqualValues(t, internal, blob.ToInternal(), "json should encode to internal type")
+		assert.Equal(t, blob, NewDataBlobFromInternal(internal), "json should decode from internal type")
+		// safe
+		internal.EncodingType = types.EncodingTypeThriftRW.Ptr()
+		blob.Encoding = common.EncodingTypeThriftRW
+		assert.EqualValues(t, internal, blob.ToInternal(), "thriftrw should encode to internal type")
+		assert.Equal(t, blob, NewDataBlobFromInternal(internal), "thriftrw should decode from internal type")
+
+		// all these panic
```

Comment:
we don't have enums for this in thrift/proto... beyond that I'm not sure why this is like this.  not sure where it's serialized / where it would matter.